BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/HeartbeatProvider.cs

```diff
@@ -89,6 +91,19 @@ public TimeSpan HeartbeatInterval
                 this.InitTimer();
             }
         }
+        
+        /// <summary>
+        /// Gets or sets the currently defined interval of the first heartbeat. Defaults to 1 second.
+        /// </summary>
+        public TimeSpan HeartbeatFirstInterval
+        {
+            get => this.firstInterval;
+            set
+            {
+                this.firstInterval = value;
```

Comment:
Add some bounds checking? At least make sure it's not negative (less than TimeSpan.Zero)

Reply:
Good idea:+1: